minimal-shadow-cljs-nodejs/src/server/main.cljs

```diff
@@ -132,78 +133,98 @@
+                                           ;; TODO
+                                           ;; ensure bot joins team-channels when they are created
+                                           (when is_member id)))))
+            message-text (decode-text-input (get-in payload [:view
+                                                             :state
+                                                             :values
+                                                             :sb-input1
+                                                             :broadcast2:text-input
+                                                             :value]))]
+      (request-updates! message-text channel-ids))))
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;; SparkBoard SlackBot server
 
-(defn response [cb body]
-  (cb nil #js {:statusCode 200 :body body}))
+(defn response [body]
+  (p/let [body-resolved body]
+    (println :responding-with body-resolved)
+    (p/resolve #js {:statusCode 200 :body (clj->js body-resolved)})))
+
+(j/defn parse-body [^:js {:as event
+                          :keys [body]
+                          {:keys [Content-Type]} :headers}]
+  (case Content-Type
+    "application/x-www-form-urlencoded" (uri/query-string->map (do #_decode-base64 body))
+    "application/json" (js->clj (js/JSON.parse body) :keywordize-keys true)
+    body))
 
-(defn handler [event _context callback]
+(defn handler [event _context]
   "Main AWS Lambda handler. Invoked by slackBot.
    See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html"
-  (println "[handler] event:" event)
-  (let [body (j/get event :body)]
-    (println "[handler] body: " body)
-    (cond
-      ;; Slack API: identification challenge
-      (some #{"challenge"} (js-keys (parse-json body)))
-      (response callback (j/get (.parse js/JSON (j/get event :body)) :challenge))
-
-      ;; Slack Event triggered (e.g. global shortcut)
-      (:payload (uri/query-string->map (decode-base64 body)))
-      (handle-modal! callback (-> body
-                                  decode-base64
-                                  uri/query-string->map
-                                  :payload
-                                  parse-json))
-      
-      :else
-      (response callback (clj->js {:action "broadcast update request to project channels"
-                                   :channels (request-updates! (-> (.parse js/JSON body)
-                                                                   (j/get-in [:event :user])
-                                                                   slack-user
-                                                                   (get "name"))
-                                                               (map :id (:slack/channels-raw @db)))})))))
+  (let [body (parse-body event)
+        request-type (cond (:challenge body) :challenge
```

Comment:
I'm not sure about this "`cond` creates keyword for later `case`" pattern

Reply:
am fine to change it. it was only the :payload variety that is weird because it doesn't indicate where it's coming from directly and i can simply add a comment for that.